### PR TITLE
Fixes 3872: Provides a better customizable application layout that

### PR DESCRIPTION
**Pull request by ehelms**
conforms to HTML5 best practices.

This change aims to provide a more customizable application layout by first
splitting the base layout, menu and content out into their own view files.
The base layout provides generic hooks and only the most basic common data
such as application javascript, stylesheets, inclusion of the menu and auth
token. Javascript inclusion is moved to the bottom of the page for page
loading efficiency. Further, all inline scripts are placed into the same
tag for browser loading efficiency. Stylesheet and generic head hooks are
split out for better readability. Lastly, this includes some updates to the
overall layout by including current HTML5 best practices for detecting and
declaring the doctype for polyfills and proper meta tags.

### CODE REVIEW

**Comment by ohadlevy** [comment]
@abenari could you please review? thanks!

File: app/views/layouts/base.html.erb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+<!DOCTYPE html>
+<!--[if lt IE 7]>      <html lang="<%= I18n.locale %>" class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
+<!--[if IE 7]>         <html lang="<%= I18n.locale %>" class="no-js lt-ie9 lt-ie8"> <![endif]-->
+<!--[if IE 8]>         <html lang="<%= I18n.locale %>" class="no-js lt-ie9"> <![endif]-->
+<!--[if gt IE 8]><!--> <html lang="<%= I18n.locale %>" class="no-js"> <!--<![endif]-->
```
Comment (ohadlevy) [inline app/views/layouts/base.html.erb:5]: whats no-js? do we use it anywhere? also, are we using the lt-\* classes anywhere?

Comment (ehelms) [inline app/views/layouts/base.html.erb:5]: These declarations serve to provide a consistent set of classes to be able to apply CSS overrides for older IE browsers. While we are not using them specifically as the moment, we may take advantage as we do more extensive browser testing. Similarly for no-js class - a standard way to apply styling for anything no-js related. This comes from https://github.com/h5bp/html5-boilerplate/blob/v4.3.0/doc/html.md#conditional-html-classes but is not strictly necessary to have.

**Comment by ohadlevy** [comment]
tested it with latency, and it feels much better!
@abenari could you do double check ?

**Comment by ehelms** [comment]
An expedited review would be nice as I am blocked by this. Thanks.

**Comment by abenari** [comment]
loading js at the bottom of the page makes too many pages look bad, that needs to be reverted. Other than that the patch is ok.

**Comment by ehelms** [comment]
Updated to move JavaScript into the header entirely.

**Pull request merged**